### PR TITLE
Adding support for libcurl version 7.4

### PR DESCRIPTION
**Pull request by vhvb1989**
fixes: https://github.com/Azure/azure-sdk-for-cpp/issues/1170

### CODE REVIEW

File: sdk/core/azure-core/CMakeLists.txt [hunk 1]
```diff
@@ -23,13 +23,30 @@ find_package(Threads REQUIRED)
 
 # min version for `CURLSSLOPT_NO_REVOKE`
 # https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
-set(CURL_MIN_REQUIRED_VERSION 7.44)
+set(CURL_MIN_REQUIRED_VERSION 7.4)
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CMakeLists.txt:26]: Based on offline discussion, we don't want to set the minimum required to a version from 2000, which is too old for the distros we support anyway (like Ubuntu 16.04).
https://curl.se/docs/releases.html

Leave it as 7.44, and let's re-evaluate if we should bump it up even higher.

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Based on offline discussion, we don't want to set the minimum required to a version from 2000, which is too old for the distros we support anyway (like Ubuntu 16.04).
https://curl.se/docs/releases.html

Leave it as 7.44, and let's re-evaluate if we should bump it up even higher.

File: sdk/core/azure-core/CMakeLists.txt [hunk 2]
```diff
@@ -23,13 +23,30 @@ find_package(Threads REQUIRED)
...
+#                                                                                              #
+#  Option `CURLSSLOPT_NO_REVOKE` will only be available for customers using version >= 7.44    #
+#                                                                                              #
+#  However, the Azure SDK for C++ can use an older version 7.4 without supporting this option   #
+#                                                                                              #
+#    #     #              #            #           #          #            #                   #
+set(OPTION_NO_REVOKE_REQUIRED_VERSION 7.44)
+if(${CURL_VERSION_STRING} VERSION_GREATER_EQUAL ${OPTION_NO_REVOKE_REQUIRED_VERSION})
+  message("The libcurl version found supports the `CURLSSLOPT_NO_REVOKE` option. Building it as part of the transport adapter options.")
+  add_compile_definitions(BUILD_SSL_OPTION_NO_REVOKE)
```
Comment (antkmsft) [inline sdk/core/azure-core/CMakeLists.txt:43]: 1. Prefix with AZ_, especially when the stuff goes into headers.
2. Even if you prefix with AZ_, when users include the file, they won't have `BUILD_SSL_OPTION_NO_REVOKE` defined when they include the header, so the idea won't work. See if you can get this moved into cpp, or if CURL defines its version as a macro, so you can make an `#ifdef` in the code, not in cmake.

**Review by antkmsft — CHANGES_REQUESTED** [review]
cmake defined macros don't work in headers. Try to avoid them regardless. Prefix macros with `AZ_`, so users are free to get the rest of the names defined as they want without interfering with our macros and vice versa.

**Comment by vhvb1989** [comment]
Closing as not to be implemented after reviewing it on stand up meeting. See previous comments for more info

**Pull request closed without merging**